Code/Tools/ProjectManager/Platform/Mac/ProjectUtils_mac.cpp

```diff
@@ -115,21 +116,64 @@ namespace O3DE::ProjectManager
                 QObject::tr("Running get_python script..."));
         }
 
-        AZ::IO::FixedMaxPath GetEditorDirectory()
+        AZ::IO::FixedMaxPath GetEditorExecutablePath(const AZ::IO::PathView& projectPath)
         {
-            AZ::IO::FixedMaxPath executableDirectory = AZ::Utils::GetExecutableDirectory();
-            AZ::IO::FixedMaxPath editorPath{ executableDirectory };
-            editorPath /= "../../../Editor.app/Contents/MacOS";
-            editorPath = editorPath.LexicallyNormal();
+            AZ::IO::FixedMaxPath editorPath;
+            AZ::IO::FixedMaxPath fixedProjectPath{ projectPath };
+
+            // First attempt to launch the Editor.exe within the project build directory if it exists
+            AZ::IO::FixedMaxPath buildPathSetregPath = fixedProjectPath
+            / AZ::SettingsRegistryInterface::DevUserRegistryFolder
+                / "Platform" / AZ_TRAIT_OS_PLATFORM_CODENAME / "build_path.setreg";
+            if (AZ::IO::SystemFile::Exists(buildPathSetregPath.c_str()))
+            {
+                AZ::SettingsRegistryImpl localRegistry;
+                // Merge the build_path.setreg into the local SettingsRegistry instance
+                if (AZ::IO::FixedMaxPath projectBuildPath;
+                    localRegistry.MergeSettingsFile(buildPathSetregPath.Native(),
+                        AZ::SettingsRegistryInterface::Format::JsonMergePatch)
+                    && localRegistry.Get(projectBuildPath.Native(), AZ::SettingsRegistryMergeUtils::ProjectBuildPath))
+                {
+                    //  local Settings Registry will be used to merge the build_path.setreg for the supplied projectPath
+                    AZ::IO::FixedMaxPath buildConfigurationPath = (fixedProjectPath / projectBuildPath).LexicallyNormal();
+
+                    // First try "<project-build-path>/bin/$<CONFIG>/Editor.app/Contents/MacOS"
+                    // Followed by "<project-build-path>/bin/$<PLATFORM>/$<CONFIG>/Editor.app/Contents/MacOS"
+                    // Directory existence is checked in this case
+                    buildConfigurationPath /= "bin";
+                    if (editorPath = (buildConfigurationPath
+                        / AZ_BUILD_CONFIGURATION_TYPE / "Editor.app/Contents/MacOS");
+                        AZ::IO::SystemFile::IsDirectory(editorPath.c_str()))
+                    {
+                        return editorPath;
```

Comment:
On Linux/Windows, this function returns the path to the Editor executable. The path being returned here should be `editorPath/"Editor"`.

Reply:
I am building the changes on Mac right now to test them out.
I'll make that change now.

Reply:
Addressed